src/ziggurat/streams.clj

```diff
@@ -92,6 +97,22 @@
         additional-tags  {:topic_name topic-entity-name}]
     (.transform stream-builder (transformer-supplier metric-namespace oldest-processed-message-in-s additional-tags) (into-array [(.name (store-supplier-builder))]))))
 
+(defn- traced-handler-fn [handler-fn channels message-payload topic-entity]
+  (let [parent-ctx (TracingKafkaUtils/extractSpanContext (:headers message-payload) tracer)
+        span (-> tracer
+                 (.buildSpan "Message-Handler")
+                 (.asChildOf parent-ctx)
+                 (.withTag (.getKey Tags/SPAN_KIND) Tags/SPAN_KIND_CONSUMER)
+                 (.withTag (.getKey Tags/COMPONENT) "lambda")
+                 (.start))]
+    (try
+      (.activate (.scopeManager tracer) span)
+      ((mapper-func handler-fn channels (:headers message-payload)) (->MessagePayload (:value message-payload) topic-entity))
```

Comment:
@roobalimsab You could have put `(:headers message-payload)` in `MessagePayload` itself. That way you would not have needed to add an extra `headers` parameter everywhere in the code.

@theanirudhvyas WDYT?

Reply:
we did not want to mess up with the existing message payload. we were not sure if the message payload will already have any headers. that is why kept it separate. if you don't see any issues, then we could merge it

Reply:
the message-payload does not have any headers, you can merge it. Also, instead of calling it headers why not call it what it exactly is - tracing-ids

Reply:
those are the kafka headers that we are passing across. hence the name